File: RxTools-library/src/main/java/com/vondear/rxtools/view/progressing/animation/interpolator/PathInterpolatorCompatApi21.java
package com.vondear.rxtools.view.progressing.animation.interpolator;

import android.annotation.TargetApi;
import android.graphics.Path;
import android.os.Build;
import android.view.animation.Interpolator;
import android.view.animation.PathInterpolator;

/**
 * @author vondear
 * API 21+ implementation for path interpolator compatibility.
 */
class PathInterpolatorCompatApi21 {

    private PathInterpolatorCompatApi21() {
        // prevent instantiation
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public static Interpolator create(Path path) {
        return new PathInterpolator(path);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public static Interpolator create(float controlX, float controlY) {
        return new PathInterpolator(controlX, controlY);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public static Interpolator create(float controlX1, float controlY1,
                                      float controlX2, float controlY2) {
        return new PathInterpolator(controlX1, controlY1, controlX2, controlY2);
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/progressing/animation/interpolator/PathInterpolatorCompatApi21.kt
package com.tamsiree.rxui.view.progressing.animation.interpolator

import android.annotation.TargetApi
import android.graphics.Path
import android.os.Build
import android.view.animation.Interpolator
import android.view.animation.PathInterpolator

/**
 * @author tamsiree
 * API 21+ implementation for path interpolator compatibility.
 */
internal object PathInterpolatorCompatApi21 {
    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    fun create(path: Path?): Interpolator {
        return PathInterpolator(path)
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    fun create(controlX: Float, controlY: Float): Interpolator {
        return PathInterpolator(controlX, controlY)
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    fun create(controlX1: Float, controlY1: Float,
               controlX2: Float, controlY2: Float): Interpolator {
        return PathInterpolator(controlX1, controlY1, controlX2, controlY2)
    }
}